### PR TITLE
Add JMS Driver

### PR DESCRIPTION
**Pull request by eolivelli**
This patch adds support to run the benchmark using the JMS API.

It includes example configuration files for Apache Kafka, Apache Pulsar and Apache Artemis.

I will follow up with the documentation.

The drivers are not bundled with the binary distribution (for instance Confluent Kafka drivers are not free), I will add instructions about how to add them to the "lib" directory.

As the JMS API do not provide "admin" operations, it is possible to delegate the initialisation of the cluster to another driver (for instance if you run JMS for Pulsar you can let the Pulsar driver to prepare the namespace and the topics)

### CODE REVIEW

**Comment by eolivelli** [comment]
@sijie @merlimat @kramasamy @k2la PTAL

File: driver-jms/kafka-jms.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+connectionFactoryConfigurationParam: '{"bootstrap.servers":"localhost:9092","client.id":"benchmarks","allow.out.of.order.acknowledge":"true","group.id":"jmsconsumergroup1"}'
+
+# Kafka JMS do not support JMS 2.0, so we can use only 1.1 API
+use20api: false
+
+# JMS API do not provide functions to admin the cluster
+# So we are going to delegate the initialization to another specific Driver
+delegateForAdminOperationsClassName: io.openmessaging.benchmark.driver.kafka.KafkaBenchmarkDriver
+
+# Pulsar specific configuration
```
Comment (k2la) [inline driver-jms/kafka-jms.yaml:33]:
```suggestion
# Kafka specific configuration
```

Comment (eolivelli) [reply driver-jms/kafka-jms.yaml]: @k2la 
good catch ! committed

**Comment by eolivelli** [comment]
@k2la thank you for your quick review

 do you have other comments ?
is the patch good to go now ?

I will send more patches in this series, this is only the initial patch to add basic support

**Comment by eolivelli** [comment]
@k2la  is there something I should add to this patch ?

**Comment by k2la** [comment]
> @k2la is there something I should add to this patch ?

No. LGTM.

File: driver-jms/pulsar-jms.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  namespacePrefix: public/default
+  topicType: persistent
+  persistence:
+    ensembleSize: 1
+    writeQuorum: 1
+    ackQuorum: 1
+    deduplicationEnabled: false
+producer:
+  blockIfQueueFull: true
+  pendingQueueSize: 10000
```
Comment (k2la) [inline driver-jms/pulsar-jms.yaml:46]: [tips] please add newline.

Comment (eolivelli) [reply driver-jms/pulsar-jms.yaml]: done, thanks

**Comment by eolivelli** [comment]
@k2la I have addressed your last comment

**Review by merlimat — APPROVED** [review]

**Pull request merged**